### ISSUE TITLE
Replace the default IoC container w/ AutoFac

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#12]
### Who

- [ ] Anonymous visitor
- [ ] Patient
- [ ] Doctor
- [ ] Content editor
- [ ] Marketer
- [ ] Administrator
- [x] Developer

### Needs what

To use advanced DI features of [AutoFac](https://autofaccn.readthedocs.io/en/latest/integration/aspnetcore.html?highlight=core).

### Why

The standard ASP.NET container [does not support](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-3.1#default-service-container-replacement) some of the required features.

### Issue expands upon

This issue expands upon another issue(s): #10 

### Blockers

This issue is blocked by the following issue(s): #`issue number`
